src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -84,8 +84,8 @@ public function getLocale(HTTPRequest $request)
             $locale = $this->getPersistLocale($request);
         }
 
-        // Check for locale in browser headers
-        if (empty($locale)) {
+        // Home page only: check for locale in browser headers
+        if (empty($locale) && $request->getURL() === '') {
```

Comment:
Makes sense. Users wanting to implement their own locale detection middleware would need to ensure it's called before this middleware, but after the InitStateMiddleware